src/rideables/MontageStack.hpp

```diff
@@ -14,35 +14,64 @@
 using namespace std;
 
 template <typename T>
-class MontageStack : public RStack<T>
+class MontageStack : public RStack<T>, Recoverable
 {
 
 private:
     struct StackNode
     {
         T data;
         StackNode *next;
-        Payload* data;
+        Payload *data;
```

Comment:
We want to place data only in persistent `payloads`. The first T-typed `data` is unnecessary here.